src/ophyd_async/epics/motion/motor.py

```diff
@@ -43,13 +88,60 @@ def __init__(self, prefix: str, name="") -> None:
         self.motor_stop = epics_signal_x(prefix + ".STOP")
         # Whether set() should complete successfully or not
         self._set_success = True
+
+        self.fly_info: Optional[FlyMotorInfo] = None
+
+        # end_position of a fly move, with run_up_distance added on.
+        self._fly_completed_position: Optional[float] = None
+
+        # Set on kickoff(), complete when motor reaches self._fly_completed_position
+        self._fly_status: Optional[WatchableAsyncStatus] = None
+
         super().__init__(name=name)
 
     def set_name(self, name: str):
         super().set_name(name)
         # Readback should be named the same as its parent in read()
         self.user_readback.set_name(name)
 
+    @AsyncStatus.wrap
+    async def prepare(self, value: FlyMotorInfo):
+        """Calculate required velocity and run-up distance, then if motor limits aren't
+        breached, move to start position minus run-up distance"""
+
+        self.fly_info = value
```

Comment:
Could: store `_fly_timeout` rather than the whole `fly_info`. If you do store the whole thing it should be private.